source/time.tex

```diff
@@ -7752,713 +7789,339 @@
 
 \begin{itemdescr}
 \pnum
 \returns
-The stored minute of \tcode{*this}.
+\tcode{m}.
 \end{itemdescr}
 
-\indexlibrarymember{seconds}{time_of_day<seconds>}%
+\indexlibrarymember{seconds}{hh_mm_ss}%
 \begin{itemdecl}
 constexpr chrono::seconds seconds() const noexcept;
 \end{itemdecl}
 
 \begin{itemdescr}
 \pnum
 \returns
-The stored second of \tcode{*this}.
+\tcode{s}.
 \end{itemdescr}
 
-\indexlibrarymember{operator precision}{time_of_day<seconds>}%
+\indexlibrarymember{subseconds}{hh_mm_ss}%
 \begin{itemdecl}
-constexpr explicit operator precision() const noexcept;
+constexpr precision subseconds() const noexcept;
 \end{itemdecl}
 
 \begin{itemdescr}
 \pnum
 \returns
-The number of seconds since midnight.
+\tcode{ss}.
 \end{itemdescr}
 
-\indexlibrarymember{to_duration}{time_of_day<seconds>}%
+\indexlibrarymember{to_duration}{hh_mm_ss}%
 \begin{itemdecl}
 constexpr precision to_duration() const noexcept;
 \end{itemdecl}
 
 \begin{itemdescr}
 \pnum
-\returns \tcode{precision\{*this\}}.
+\returns
+If \tcode{is_neg}, returns \tcode{-(h + m + s + ss)},
+otherwise returns \tcode{h + m + s + ss}.
 \end{itemdescr}
 
-\indexlibrarymember{make24}{time_of_day<seconds>}%
+\indexlibrarymember{operator precision}{hh_mm_ss}%
 \begin{itemdecl}
-constexpr void make24() noexcept;
+constexpr explicit operator precision() const noexcept;
 \end{itemdecl}
 
 \begin{itemdescr}
 \pnum
-\effects
-If \tcode{*this} is a 12-hour time,
-converts to a 24-hour time.
-Otherwise, no effects.
+\returns
+\tcode{to_duration()}.
 \end{itemdescr}
 
-\indexlibrarymember{make12}{time_of_day<seconds>}%
+\rSec2[time.hms.nonmembers]{Non-members}
+
 \begin{itemdecl}
-constexpr void make12() noexcept;
+template <class charT, class traits, class Duration>
+basic_ostream<charT, traits>&
+operator<<(basic_ostream<charT, traits>& os, hh_mm_ss<Duration> const& hms);
 \end{itemdecl}
 
 \begin{itemdescr}
 \pnum
 \effects
-If \tcode{*this} is a 24-hour time,
-converts to a 12-hour time.
-Otherwise, no effects.
-\end{itemdescr}
-
-\rSec2[time.tod.subsecond]{Sub-second precision}
-\indexlibrary{\idxcode{time_of_day<\placeholder{sub-second duration}>}}
-
+Equivalent to:
 \begin{codeblock}
-namespace std::chrono {
-  template<class Rep, class Period>
-  class time_of_day<duration<Rep, Period>> {
-  public:
-    using precision = duration<Rep, Period>;
-
-    time_of_day() = default;
-    constexpr explicit time_of_day(precision since_midnight) noexcept;
-
-    constexpr chrono::hours     hours()      const noexcept;
-    constexpr chrono::minutes   minutes()    const noexcept;
-    constexpr chrono::seconds   seconds()    const noexcept;
-    constexpr precision subseconds() const noexcept;
-
-    constexpr explicit operator  precision()   const noexcept;
-    constexpr          precision to_duration() const noexcept;
-
-    constexpr void make24() noexcept;
-    constexpr void make12() noexcept;
-  };
-}
+if (hms.is_negative())
+  os << @\placeholder{STATICALLY-WIDEN}@<charT>("-");
```

Comment:
This needs to be part of the format string; otherwise we'd be applying the the stream width/fill to this single character instead of the whole output.

Reply:
Agreed, but this was already merged (and a fix doesn't seem editorial to me). @timsong-cpp, please post to the LWG reflector and ask for a new library issue.

Reply:
Well, the paper itself doesn't use this specification (it's not fmt-based), and I think you can at least arguably read the original spec as doing the right thing.

The translation to fmt that made the spec unambiguously incorrect is the result of an editorial change (1a37c22).

Reply:
@zygoloid, there is a problem here.

Reply:
Hmm. The original wording was ambiguous in this regard. Does "the output is preceded by" mean a single stream operation or two?

"Outputs to `os` according to the format `%T`" doesn't mean anything any more after the `std::format` for \<chrono> changes, so reverting this is also wrong.

I was planning to mail LWG for feedback on this change anyway.

Reply:
Fixed in 584a87ec1d48862b9e68a269d0a5eb7b05d6999d after consultation with LWG.